### PR TITLE
:safety_vest: added codeql action

### PR DESCRIPTION
**Pull request by stiidk**
We want to try out semantic code analysis by codeql

### CODE REVIEW

File: .github/workflows/codeql.yaml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    # CodeQL runs on ubuntu-latest, windows-latest, and macos-latest
+    runs-on: ubuntu-latest
+
+    permissions:
+      # required for all workflows
+      security-events: write
+
+      # only required for workflows in private repositories
+      actions: read
+      contents: read
```
Comment (blacksails) [inline .github/workflows/codeql.yaml:18]: this is not needed

File: .github/workflows/codeql.yaml [hunk 2]
```diff
@@ -0,0 +1,32 @@
+name: "Code Scanning - Action"
+
+on:
+  schedule:
+    - cron: '0 4 * * *'
```
Comment (blacksails) [inline .github/workflows/codeql.yaml:5]: Maybe we should do it on any master commit while we test?

**Review by blacksails — CHANGES_REQUESTED** [review]

File: .github/workflows/codeql.yaml [hunk 3]
```diff
@@ -0,0 +1,32 @@
+name: "Code Scanning - Action"
```
Comment (simlau) [inline .github/workflows/codeql.yaml:15]: missing license

File: .github/workflows/codeql.yaml [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+on:
+  schedule:
+    - cron: '0 4 * * *'
+
+jobs:
+  CodeQL-Build:
+    # CodeQL runs on ubuntu-latest, windows-latest, and macos-latest
```
Comment (simlau) [inline .github/workflows/codeql.yaml:25]: remove useless comments

**Review by blacksails — APPROVED** [review]

**Review by simlau — CHANGES_REQUESTED** [review]

**Review by simlau — APPROVED** [review]

**Pull request merged**